src/managers/GroupManager.ts

```diff
@@ -82,6 +85,44 @@ export default class GroupManager extends BaseManager<Group> implements GroupMan
         return group
     }
 
+    /**
+     * Joins a group.
+     *
+     * @param inviteLinkOrGroupID The group invite link or group ID.
+     * @param shareToken The group's share token.
+     * @returns The joined group.
+     */
+    join(inviteLink: string): Promise<Group>
+    join(groupID: string, shareToken: string): Promise<Group>
+    public async join(inviteLinkOrGroupID: string, shareToken?: string): Promise<Group> {
+        if (shareToken !== undefined) {
+            return await this.joinWithToken(inviteLinkOrGroupID, shareToken)
+        } else {
+            const urlPath = new URL(inviteLinkOrGroupID).pathname
```

Comment:
good call on using the `URL` API here. since we're parsing it anyway, why not check the domain too?